baseapp/baseapp.go

```diff
@@ -119,10 +119,15 @@ type BaseApp struct {
 	//
 	// - finalizeBlockState: Used for FinalizeBlock, which is set based on the
 	// previous block's state. This state is committed.
+	//
+	// NOTE: The states should be accessed via getter and setter to avoid race conditions.
+	// - getter: getState
+	// - setter: setState and clearState
 	checkState           *state
 	prepareProposalState *state
 	processProposalState *state
 	finalizeBlockState   *state
```

Comment:
we should probably move these into a struct or internal object in a way that forces devs to use the `get` methods

Reply:
like maybe a `state` package in `baseapp/state` with just public `Get`